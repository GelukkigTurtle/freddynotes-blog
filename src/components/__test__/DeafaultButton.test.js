import React from 'react'
import renderer from 'react-test-renderer'

import { DefaultButton } from '../ButtonComponent'

describe('DefaultButton', () => {
  it('Render Default Option', () => {
    const defaultButton = renderer
      .create(<DefaultButton>Button</DefaultButton>)
      .toJSON()

    expect(defaultButton).toMatchSnapshot()
  })

  it('Render To Option', () => {
    const defaultButton = renderer
      .create(<DefaultButton to={'http://www.naver.com'}>To Btn</DefaultButton>)
      .toJSON()

    expect(defaultButton).toMatchSnapshot()
  })

  it('Render customClass Option', () => {
    const defaultButton = renderer
      .create(
        <DefaultButton customClass={'snsBtn'}>CustomClass Btn</DefaultButton>
      )
      .toJSON()

    expect(defaultButton).toMatchSnapshot()
  })
})
