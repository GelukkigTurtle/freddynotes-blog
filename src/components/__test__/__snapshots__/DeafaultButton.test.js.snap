// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultButton Render Default Option 1`] = `
<a
  className="defaultClass buttonContainer"
  href="/https:/gatsby-sseon-starter.netlify.com/"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  Button
</a>
`;

exports[`DefaultButton Render To Option 1`] = `
<a
  className="defaultClass buttonContainer"
  href="/http:/www.naver.com"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  To Btn
</a>
`;

exports[`DefaultButton Render customClass Option 1`] = `
<a
  className="snsBtn"
  href="/https:/gatsby-sseon-starter.netlify.com/"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  CustomClass Btn
</a>
`;
